xoa/data_source_vms_test.go

```diff
@@ -0,0 +1,51 @@
+package xoa
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccXenorchestraDataSource_vms(t *testing.T) {
+	resourceName := "data.xenorchestra_vms.vms"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccXenorchestraDataSourceVmsConfig(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckXenorchestraDataSourceVms(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "sum")),
```

Comment:
Can we add an assertion here to make sure `vms.#` is non zero? We should be able to use [this regex function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/resource#TestMatchResourceAttr) to accomplish that.

```
TestMatchResourceAttr(resourceName, "vms.#", regexp.MustCompile("^[1-9][0-9]*$")
```

Reply:
:+1: